lib/next_ls/check.ex

```diff
@@ -0,0 +1,30 @@
+defmodule NextLS.Check do
+  @moduledoc """
+  Data structures used for NextLS checks.
+  """
+
+  @doc """
+  Data structure that holds information related to an instance of a check found by NextLS.
+  """
+  defstruct [:check, :diagnostic, :uri, :document]
+
+  def new(opts) do
+    opts = Keyword.update!(opts, :check, &String.to_existing_atom/1)
+    struct(__MODULE__, opts)
+  end
+
+  defimpl NextLS.CodeActionable do
+    alias NextLS.CodeAction
```

Comment:
I imagine we'll have several CodeActions, some basic ones for example:
`NextLS.CodeAction.UnusedVar`
`NextLS.CodeAction.AddRequire`

And each will have a `new/1` function that converts a diagnostic/check/placeholder struct to the code action which will contain the necessary TextEdits inside for the client.